### ISSUE TITLE
Questions regarding aria-haspopup

### ISSUE THREAD

**Issue by kristofferlium** [w3c/aria#795]
I have some questions regarding the concept of `aria-haspopup` and perhaps various nuances of the usage.

1. What is the basic concept for the usage of `aria-haspopup`? E.g. for a user such as myself (I’m blind and navigate mostly with VoiceOver), an element being visually «on top of other content» is nothing I can relate to. Unless, that is, if the popup restricts navigation outside of itself.

2. According to the WAI-ARIA 1.1 spec `aria-haspopup` can take on a range of different values. But from what I can tell, these values seem to be different «types». For example I think that «dialog» describes the container while «tree» describes the content. So what can we infer of the (behavior of the) container if we use «tree» as the value?

3. The HTML spec describes that the `<dialog>` element should contain navigation while ARIA says simply that the element with `role=dialog` (non-modal) should have at least one focusable element (see semi-related issue https://github.com/w3c/aria/issues/794). What is the intended behavior of the container that the element that has aria-haspopup references and does this change depending on the values set?

**Comment by jnurthen** [comment]
@kristofferlium I hope these answers help. For more authoring guidance may I suggest looking at the ARIA Authoring Practices Guide and filing issues there where more clarifications are needed. In the APG the [menu button](https://w3c.github.io/aria-practices/#menubutton) pattern shows examples for aria-haspopup=menu and the [combobox](https://w3c.github.io/aria-practices/#combobox) pattern for aria-haspopup=tree/grid/listbox/dialog

1. It indicates the type of control which will open when you interact with the element. It should be helpful to know this although of course the opened content should expose this same semantic.
2. there may be no seperate container for tree other than the control itself. Tree could be used when opening a combo box and the tree is the element from which you select a value in the combo.
3. I don't think I understand this question. If the above don't clarify can you please rephrase it.

**Comment by jnurthen** [comment]
Marking closed. Please re-open if necessary.
